examples/hotrod/services/route/client.go

```diff
@@ -30,20 +30,18 @@ import (
 
 // Client is a remote client that implements route.Interface
 type Client struct {
-	tracer   opentracing.Tracer
 	logger   log.Factory
 	client   *tracing.HTTPClient
 	hostPort string
 }
 
 // NewClient creates a new route.Client
-func NewClient(tracer opentracing.Tracer, logger log.Factory, hostPort string) *Client {
+func NewClient(tracer trace.TracerProvider, logger log.Factory, hostPort string) *Client {
 	return &Client{
-		tracer: tracer,
 		logger: logger,
 		client: &tracing.HTTPClient{
-			Client: &http.Client{Transport: &nethttp.Transport{}},
-			Tracer: tracer,
+			Client:         &http.Client{Transport: &otelhttp.Transport{}},
```

Comment:
It doesn't seem like there are situations where you would NOT pass this value. Therefore, I would recommend creating a constructor `tracing.NewHTTPClient(tracerProvider)` to minimize the call at the call sites